js/engine/Main.ml

```diff
@@ -71,22 +71,44 @@ let _ =
          | Some lang -> Js.some (Js.string (Lang.to_lowercase_alnum lang))
          | None -> Js.null
 
-       method execute language rule_file source_file : string =
+       method execute language rule_file root source_files : string =
+         let xlang = Xlang.of_string (Js.to_string language) in
+         let rules_and_errors =
```

Comment:
If this is mostly a copy paste of some code in Run_semgrep.ml or something, please add a (* coupling:  *) comment
explaining the copy-paste and how it differs from the original.